# Imports

In [2]:
import pandas as pd
import sys
sys.path.append('../src/')

from data_ingest import DataIngest

# Loading data

In [3]:
di = DataIngest()
df = di.load_data()

2022-11-17 20:34.50 [info     ] /home/matheus/Documents/repos/customers_intention_purchasing/src/../data/raw
2022-11-17 20:34.50 [info     ] Starting loading data...
2022-11-17 20:34.50 [info     ] Success loading data...


# Knowing the data

| Nome da coluna          | Descrição                                                                                                        |
|-------------------------|------------------------------------------------------------------------------------------------------------------|
| Administrative          | Número de páginas administrativas que o usuário visitou                                                          |
| Administrative_Duration | Total de tempo gasto na categoria da página administrativa                                                       |
| Informational           | Número de páginas informativa que o usuário visitou                                                              |
| Informational_Duration  | Total de tempo gasto na categoria da página informativa                                                          |
| ProductRelated          | Número de páginas relacionadas à produtos que o usuário visitou                                                  |
| ProductRelated_Duration | Total de tempo gasto na categoria da página relacionadas à produtos                                              |
| BounceRates             | A porcentagem de visitantes que entram no site por meio dessa página e saem sem acionar nenhuma tarefa adicional |
| ExitRates               | A porcentagem de visualizações de página no site que terminam nessa página específica.                           |
| PageValues              | O valor médio da página em média sobre o valor da página de destino e/ou a conclusão de um eCommerce             |
| SpecialDay              | Este valor representa a proximidade da data de navegação para dias ou feriados especiais                         |
| Month                   | Mês da jornada do usuário                                                                                        |
| OperatingSystems        | Indica o sistema operacional utilizado pelo usuário em sua jornada                                               |
| Browser                 | Indica o browser utilizado pelo usuário em sua jornada                                                           |
| Region                  | Indica a região de acesso onde o usário acessou o website                                                        |
| TrafficType             | Indica qual tipo do trafégo foi utilizado pelo usuário para acesso ao website                                    |
| VisitorType             | Indica se o usuário é um retornante ou novo                                                                      |
| Weekend                 | Indica se no dia do acesso do usuário foi um dia de final de semana                                              |
| Revenue                 | Resultado da navegação do usuário se foi realizado uma compra ou não                                             |

## General data overview

In [4]:
df.head()

Administrative  Administrative_Duration  Informational  \
0               0                      0.0              0   
1               0                      0.0              0   
2               0                      0.0              0   
3               0                      0.0              0   
4               0                      0.0              0   

   Informational_Duration  ProductRelated  ProductRelated_Duration  \
0                     0.0               1                 0.000000   
1                     0.0               2                64.000000   
2                     0.0               1                 0.000000   
3                     0.0               2                 2.666667   
4                     0.0              10               627.500000   

   BounceRates  ExitRates  PageValues  SpecialDay Month  OperatingSystems  \
0         0.20       0.20         0.0         0.0   Feb                 1   
1         0.00       0.10         0.0         0.0   Feb                 2   
2         0.20       0.20         0.0         0.0   Feb                 4   
3         0.05       0.14         0.0         0.0   Feb                 3   
4         0.02       0.05         0.0         0.0   Feb                 3   

   Browser  Region  TrafficType        VisitorType  Weekend  Revenue  
0        1       1            1  Returning_Visitor    False    False  
1        2       1            2  Returning_Visitor    False    False  
2        1       9            3  Returning_Visitor    False    False  
3        2       2            4  Returning_Visitor    False    False  
4        3       1            4  Returning_Visitor     True    False

Com os dados acima podemos visualizar de forma geral os dados que iremos trabalhar nos próximos passos.

## Dimensionality

In [5]:
df.shape

(12330, 18)

Como dito no README, temos disponibilizados 12330 linhas e com 18 colunas.

In [12]:
df.Revenue.value_counts(normalize=True)

False    0.845255
True     0.154745
Name: Revenue, dtype: float64

Podemos perceber da nossa variável alvo, ```Revenue```, que temos a questão dos dados estarem desbalanceados, necessitando assim um olhar mais atento ao decorrer do projeto.

## Data description

In [11]:
df.describe(include='all')

Administrative  Administrative_Duration  Informational  \
count     12330.000000             12330.000000   12330.000000   
unique             NaN                      NaN            NaN   
top                NaN                      NaN            NaN   
freq               NaN                      NaN            NaN   
mean          2.315166                80.818611       0.503569   
std           3.321784               176.779107       1.270156   
min           0.000000                 0.000000       0.000000   
25%           0.000000                 0.000000       0.000000   
50%           1.000000                 7.500000       0.000000   
75%           4.000000                93.256250       0.000000   
max          27.000000              3398.750000      24.000000   

        Informational_Duration  ProductRelated  ProductRelated_Duration  \
count             12330.000000    12330.000000             12330.000000   
unique                     NaN             NaN                      NaN   
top                        NaN             NaN                      NaN   
freq                       NaN             NaN                      NaN   
mean                 34.472398       31.731468              1194.746220   
std                 140.749294       44.475503              1913.669288   
min                   0.000000        0.000000                 0.000000   
25%                   0.000000        7.000000               184.137500   
50%                   0.000000       18.000000               598.936905   
75%                   0.000000       38.000000              1464.157214   
max                2549.375000      705.000000             63973.522230   

         BounceRates     ExitRates    PageValues    SpecialDay  Month  \
count   12330.000000  12330.000000  12330.000000  12330.000000  12330   
unique           NaN           NaN           NaN           NaN     10   
top              NaN           NaN           NaN           NaN    May   
freq             NaN           NaN           NaN           NaN   3364   
mean        0.022191      0.043073      5.889258      0.061427    NaN   
std         0.048488      0.048597     18.568437      0.198917    NaN   
min         0.000000      0.000000      0.000000      0.000000    NaN   
25%         0.000000      0.014286      0.000000      0.000000    NaN   
50%         0.003112      0.025156      0.000000      0.000000    NaN   
75%         0.016813      0.050000      0.000000      0.000000    NaN   
max         0.200000      0.200000    361.763742      1.000000    NaN   

        OperatingSystems       Browser        Region   TrafficType  \
count       12330.000000  12330.000000  12330.000000  12330.000000   
unique               NaN           NaN           NaN           NaN   
top                  NaN           NaN           NaN           NaN   
freq                 NaN           NaN           NaN           NaN   
mean            2.124006      2.357097      3.147364      4.069586   
std             0.911325      1.717277      2.401591      4.025169   
min             1.000000      1.000000      1.000000      1.000000   
25%             2.000000      2.000000      1.000000      2.000000   
50%             2.000000      2.000000      3.000000      2.000000   
75%             3.000000      2.000000      4.000000      4.000000   
max             8.000000     13.000000      9.000000     20.000000   

              VisitorType Weekend Revenue  
count               12330   12330   12330  
unique                  3       2       2  
top     Returning_Visitor   False   False  
freq                10551    9462   10422  
mean                  NaN     NaN     NaN  
std                   NaN     NaN     NaN  
min                   NaN     NaN     NaN  
25%                   NaN     NaN     NaN  
50%                   NaN     NaN     NaN  
75%                   NaN     NaN     NaN  
max                   NaN     NaN     NaN

Alguns insights da descrição dos dados:
- Existem usuários que ficam em páginas administrativa e informativa bastante tempo
- Maio é o mês mais frequente dos acessos dos usuários
- Maioria dos usuários que acessaram ao decorrer desse período são usuários que retornaram
- Maioria dos usuários não acessam o site no fim de semana

In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12330 entries, 0 to 12329
Data columns (total 18 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Administrative           12330 non-null  int64  
 1   Administrative_Duration  12330 non-null  float64
 2   Informational            12330 non-null  int64  
 3   Informational_Duration   12330 non-null  float64
 4   ProductRelated           12330 non-null  int64  
 5   ProductRelated_Duration  12330 non-null  float64
 6   BounceRates              12330 non-null  float64
 7   ExitRates                12330 non-null  float64
 8   PageValues               12330 non-null  float64
 9   SpecialDay               12330 non-null  float64
 10  Month                    12330 non-null  object 
 11  OperatingSystems         12330 non-null  int64  
 12  Browser                  12330 non-null  int64  
 13  Region                   12330 non-null  int64  
 14  TrafficType           

In [8]:
df.isnull().sum()

Administrative             0
Administrative_Duration    0
Informational              0
Informational_Duration     0
ProductRelated             0
ProductRelated_Duration    0
BounceRates                0
ExitRates                  0
PageValues                 0
SpecialDay                 0
Month                      0
OperatingSystems           0
Browser                    0
Region                     0
TrafficType                0
VisitorType                0
Weekend                    0
Revenue                    0
dtype: int64

Os retornos acima indicam que não temos dados faltantes, pois iremos adotar a premissa que os dados estão corretamente limpos.